src/components/FileManager.vue

```
@@ -0,0 +1,290 @@
+              </div>
+          </div>
+
+          <div
+            v-for="document in orderedDocuments"
+            v-bind:key="document.name"
+            class="document"
+            >
+              <div
+                class="document-delete"
+                @click="deleteDocument(document)"
+                >
+                  <DeleteIcon class="delete-icon" />
+              </div>
+              <div class="document-image">
+                    <img v-if="document.thumbnail" :src="document.thumbnail">
+                    <GenericFileImage v-else class="generic-file-image" />
+              </div>
+              <div class="document-name">
+                {{ document.name }}
+              </div>
+            </div>
+
+      </div>
+    </div>
+    </drop-area>
+</template>
+
+<script>
+import _ from 'lodash'
+import API from '../api.js'
+import DropArea from './DropArea.vue'
+import DeleteIcon from './../assets/cancel.svg'
+import GenericFileImage from './../assets/file.svg'
+
+export default {
+  name: 'filemanager',
+  components: {
+    DropArea,
+    DeleteIcon,
+    GenericFileImage
+  },
+  props: ['apiUrl', 'maxUploadRetries'],
+  data () {
+    return {
+      uploads: [],
+      documents: []
+    }
+  },
+  computed: {
+    orderedDocuments () {
+      return _.orderBy(this.documents, 'name', 'asc')
+    }
+  },
+  created () {
+    if (this.maxUploadRetries === undefined || this.maxUploadRetries === null) {
+      this.maxUploadRetries = 3
+    }
+    API.setUrlAPI(this.apiUrl)
+    this.getDocuments()
+    let self = this
+    setInterval(function () {
+      self.getDocuments()
+    }, 30000)
+  },
+  methods: {
+    deleteDocument (document) {
+      const res = confirm('Are you sure to delete [' + document.name + ']?')
+      if (res) {
+        API.deleteFile(document.name)
+          .then(data => {
+            this.getDocuments()
+          })
+          .catch(error => {
+            console.log(error)
```

error is dropped silently